src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -0,0 +1,369 @@
+                "OrganizationExternalDatabaseTable ${tableFQN.fullQualifiedNameAsString} already exists")
+        aclKeyReservations.reserveIdAndValidateType(table, tableFQN::getFullQualifiedNameAsString)
+
+        val tableAclKey = AclKey(orgId, table.id)
+        authorizationManager.setSecurableObjectType(tableAclKey, SecurableObjectType.OrganizationExternalDatabaseTable)
+        authorizationManager.addPermission(tableAclKey, principal, EnumSet.allOf(Permission::class.java))
+        //eventBus?
+
+        return table.id
+    }
+
+    fun createOrganizationExternalDatabaseColumn(orgId: UUID, column: OrganizationExternalDatabaseColumn): UUID {
+        val principal = Principals.getCurrentUser()
+        Principals.ensureUser(principal)
+
+        checkState(organizationExternalDatabaseTables[column.tableId] != null,
+                "OrganizationExternalDatabaseColumn ${column.name} belongs to a table that does not exist")
+        val columnFQN = FullQualifiedName(column.tableId.toString(), column.name)
+        checkState(organizationExternalDatabaseColumns.putIfAbsent(column.id, column) == null,
+                "OrganizationExternalDatabaseColumn ${columnFQN.fullQualifiedNameAsString} already exists")
+        aclKeyReservations.reserveIdAndValidateType(column, columnFQN::getFullQualifiedNameAsString)
+
+        val columnAclKey = AclKey(orgId, column.tableId, column.id)
+        authorizationManager.setSecurableObjectType(columnAclKey, SecurableObjectType.OrganizationExternalDatabaseColumn)
+        authorizationManager.addPermission(columnAclKey, principal, EnumSet.allOf(Permission::class.java))
+
+        return column.id
+    }
+
+    /*GET*/
+    fun getOrganizationIds(): Set<UUID> {
+        val orgIds = mutableSetOf<UUID>()
+        hds.connection.createStatement().use {
+            val rs = it.executeQuery("SELECT id FROM organizations")
+            while (rs.next()) {
+                orgIds.add(organizationId(rs))
+            }
+        }
+        return orgIds
+    }
+
+    fun getOrganizationExternalDatabaseTable(tableId: UUID): OrganizationExternalDatabaseTable {
+        return organizationExternalDatabaseTables[tableId]!!
+    }
+
+    fun getOrganizationExternalDatabaseColumn(columnId: UUID): OrganizationExternalDatabaseColumn {
+        return organizationExternalDatabaseColumns[columnId]!!
+    }
+
+    /*DELETE*/
+    fun deleteOrganizationExternalDatabaseTables(orgId: UUID, tableIds: Set<UUID>) {
+        tableIds.forEach { deleteOrganizationExternalDatabaseTable(orgId, it) }
+    }
+
+    fun deleteOrganizationExternalDatabaseTable(orgId: UUID, tableId: UUID) {
+        organizationExternalDatabaseTables.remove(tableId) //TODO make this a set so we can batch delete, entry processor?
+        aclKeyReservations.release(tableId)
+
+        //delete columns that belonged to this table
+        val belongsToDeletedTable = Predicates.equal("tableId", tableId)
```

Comment:
Also, there should be an index for this probably too + would use the constant for it (you can move it to the companion class as `@JvmStatic val ...` to be able to use it as `OrganizationExternalDatabaseTableMapstore.TABLE_ID_INDEX`)

Reply:
Ope, I do have that index in the mapstore and did not use it. Ty for the catch!

Reply:
Ignore the first part of the comment, missed, that TABLE_ID_INDEX was already added to the mapstore.. So just basically using the constant here is preferred